src/components/Conversation.tsx

```diff
@@ -1,26 +1,73 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import Message from './Message';
 import { Message as MessageType } from '../types/Message';
+import { Button } from '@mui/material';
 
 interface ConversationProps {
   messages: MessageType[];
 }
 
 const Conversation: React.FC<ConversationProps> = ({ messages }) => {
+  const [selectedMessageId, setSelectedMessageId] = useState<string | null>(null);
+  const [childIndex, setChildIndex] = useState<{ [key: string]: number }>({});
+  const [selectedChildId, setSelectedChildId] = useState<string | null>(null);
+
+  useEffect(() => {
+    const rootChildren = messages.filter(message => message.parentId === null);
+    if (rootChildren.length > 0) {
+      setSelectedMessageId(rootChildren[0].id);
+    }
+  }, [messages]);
+
+  useEffect(() => {
+    if (selectedMessageId) {
+      const children = messages.filter(message => message.parentId === selectedMessageId);
+      if (children.length > 0) {
+        setSelectedChildId(children[0].id);
+        setChildIndex(prev => ({ ...prev, [selectedMessageId]: 0 }));
+      }
+    }
+  }, [selectedMessageId, messages]);
+
+  const handleSelectMessage = (messageId: string) => {
+    setSelectedMessageId(messageId);
+    setChildIndex({});
+    setSelectedChildId(null);
+  };
+
   const renderMessages = (messages: MessageType[], parentId: string | null = null): JSX.Element[] => {
-    return messages
-      .filter(message => message.parentId === parentId)
-      .map(message => (
-        <div key={message.id}>
-          <Message {...message} />
-          <div style={{ marginLeft: '20px' }}>
-            {renderMessages(messages, message.id)}
+    const filteredMessages = messages.filter(message => message.parentId === parentId);
+    if (filteredMessages.length === 0) return [];
+
+    const currentIndex = childIndex[parentId || 'root'] || 0;
+    const currentMessage = filteredMessages[currentIndex];
+
+    return [
+      <div key={currentMessage.id}>
+        <Message {...currentMessage} onClick={() => handleSelectMessage(currentMessage.id)} />
+        {selectedMessageId === currentMessage.id && (
+          <div>
+            {renderMessages(messages, currentMessage.id)}
           </div>
+        )}
+        <div style={{ display: 'flex', justifyContent: 'flex-end', marginTop: '8px' }}>
+          <Button
+            onClick={() => setChildIndex(prev => ({ ...prev, [parentId || 'root']: (currentIndex - 1 + filteredMessages.length) % filteredMessages.length }))}
+            disabled={filteredMessages.length <= 1}
+          >
+            &lt;
+          </Button>
+          <Button
+            onClick={() => setChildIndex(prev => ({ ...prev, [parentId || 'root']: (currentIndex + 1) % filteredMessages.length }))}
+            disabled={filteredMessages.length <= 1}
+          >
+            &gt;
```

Comment:
Is there not better material UI component for this? Also if possible I'd like the selectors to be injected into the message component itself so it could decide how to render them. But we can handle the logic here. Is that possible?